algorithms/linfa-bayes/src/gaussian_nb.rs

```diff
@@ -230,14 +230,24 @@ pub struct GaussianNb<F, L> {
     class_info: HashMap<L, GaussianClassInfo<F>>,
 }
 
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, PartialEq)]
 struct GaussianClassInfo<F> {
     class_count: usize,
     prior: F,
     theta: Array1<F>,
     sigma: Array1<F>,
 }
 
+impl<'a, F, L> PartialEq for GaussianNb<F, L>
+where
+    F: Float,
+    L: Label + Ord,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.class_info == other.class_info
+    }
+}
```

Comment:
This impl can be derived (the generic bounds you specified here will be generated by the `derive` anyways).

Reply:
Doing so throws a compilation error E0369, specifically `error[E0369]: binary operation '==' cannot be applied to type 'HashMap<L, GaussianClassInfo<F>>'`, wasn't sure how to fix that

Reply:
This is happening because `HashMap`'s `PartialEq` impl requires `L: Eq`, but the derive only inserts the `L: PartialEq` bound. There's no way around this, so the manual impl is required. Just change the bounds to `F: PartialEq, L: Eq`.

Reply:
Ok, added `Eq`, and additionally `Hash` as it was required by `HashMap<K, V>` impl